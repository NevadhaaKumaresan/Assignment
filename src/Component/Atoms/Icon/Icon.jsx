import React from 'react';
import { DeleteTwoTone, CheckCircleTwoTone } from '@ant-design/icons';
import PropTypes from 'prop-types';  

export const DeleteIcon = (props) => {
    return(
        <DeleteTwoTone data-testid="delete-task" twoToneColor="#eb2f96"  onClick={() => props.handleDelete(props.obj)}/>
        )
};

export const CompleteIcon = (props) => {
    return(
        <CheckCircleTwoTone data-testid="complete" twoToneColor="#52c41a" onClick={() => props.handleComplete(props.obj)}/>
        )
};

DeleteIcon.propTypes = {  
    handleDelete: PropTypes.func,
    obj: PropTypes.object
}  
CompleteIcon.propTypes = {
    handleComplete: PropTypes.func,
    obj: PropTypes.object
}

