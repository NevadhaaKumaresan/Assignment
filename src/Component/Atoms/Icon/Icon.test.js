/* eslint-disable no-undef */
import React from 'react';
import { DeleteIcon , CompleteIcon } from './Icon';
import { screen, fireEvent, render } from '@testing-library/react';

test('OnClick of delete icon, func should be called', () => {
    const obj= {
        "userId": 1,
        "id": 1,
        "title": "delectus aut autem",
        "completed": false
      };
    const handleDelete = jest.fn();
    render(<DeleteIcon obj={obj} handleDelete={() => handleDelete(obj)}/>);
    const deleteTask = screen.getByTestId("delete-task");
    fireEvent.click(deleteTask);
    expect(handleDelete).toHaveBeenCalledTimes(1);
  })

  test('OnClick of Complete icon, func should be called', () => {
    const obj= {
        "userId": 1,
        "id": 1,
        "title": "delectus aut autem",
        "completed": false
      };
    const handleComplete = jest.fn();
    render(<CompleteIcon obj={obj} handleComplete={() => handleComplete(obj)}/>);
    const moveToComplete = screen.getByTestId("complete");
    fireEvent.click(moveToComplete);
    expect(handleComplete).toHaveBeenCalledTimes(1);
  })
  