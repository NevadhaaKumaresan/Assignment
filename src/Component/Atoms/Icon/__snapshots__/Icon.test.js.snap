// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<span
  aria-label="delete"
  className="anticon anticon-delete"
  onClick={[Function]}
  role="img"
  tabIndex={-1}
>
  <svg
    aria-hidden="true"
    data-icon="delete"
    fill="currentColor"
    focusable="false"
    height="1em"
    viewBox="64 64 896 896"
    width="1em"
  >
    <path
      d="M292.7 840h438.6l24.2-512h-487z"
      fill="#fff0f6"
    />
    <path
      d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-504-72h304v72H360v-72zm371.3 656H292.7l-24.2-512h487l-24.2 512z"
      fill="#eb2f96"
    />
  </svg>
</span>
`;

exports[`matches snapshot 2`] = `
<span
  aria-label="check-circle"
  className="anticon anticon-check-circle"
  onClick={[Function]}
  role="img"
  tabIndex={-1}
>
  <svg
    aria-hidden="true"
    data-icon="check-circle"
    fill="currentColor"
    focusable="false"
    height="1em"
    viewBox="64 64 896 896"
    width="1em"
  >
    <path
      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
      fill="#52c41a"
    />
    <path
      d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm193.4 225.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.3 0 19.9 5 25.9 13.3l71.2 98.8 157.2-218c6-8.4 15.7-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.4 12.7z"
      fill="#f6ffed"
    />
    <path
      d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0051.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
      fill="#52c41a"
    />
  </svg>
</span>
`;
